### PR TITLE
Run quick tests in GitHub Linux CI

### PR DESCRIPTION
**Pull request by masterleinad**
Addressing https://github.com/dealii/dealii/pull/13466#issue-1154623729.

### CODE REVIEW

File: .github/workflows/linux.yml [hunk 1]
```diff
@@ -36,6 +36,8 @@ jobs:
     - name: build
       run: |
         make -j 2
+    - name: test
```
Comment (drwells) [inline .github/workflows/linux.yml:39]: could we rename these to `quicktests` or something similar? I'd prefer to disambiguate between these and the test suite.

**Comment by drwells** [comment]
It's alarming that the cuda quick test fails on the cuda debug tester - we should have turned this on earlier.

**Comment by masterleinad** [comment]
> It's alarming that the cuda quick test fails on the cuda debug tester - we should have turned this on earlier.

Not really. The machine doesn't actually have a `NVidia` GPU so we can't run anything that uses `CUDA`. I have turned it off for now again.

**Comment by marcfehling** [comment]
```
1/7 Test #2: affinity.release .................***Failed    6.17 sec
/home/runner/work/dealii/dealii/tests/quick_tests/affinity.cc: In function ‘bool getaffinity(unsigned int&, unsigned int&)’:
/home/runner/work/dealii/dealii/tests/quick_tests/affinity.cc:53:42: error: dereferencing type-punned pointer will break strict-aliasing rules [-Werror=strict-aliasing]
   mask = *reinterpret_cast<int *>(&my_set);
                                          ^
cc1plus: all warnings being treated as errors
```

File: .github/workflows/linux.yml [hunk 2]
```diff
@@ -222,3 +223,5 @@ jobs:
       run: |
         source /opt/intel/oneapi/setvars.sh
         make -j 2
+    - name: quicktest
+      run: make test
```
Comment (marcfehling) [inline .github/workflows/linux.yml:227]:
```suggestion
    - name: quicktest
      run: |
        source /opt/intel/oneapi/setvars.sh
        make test
```
You need to set the environmental variables again. I *think* this should solve all issues related to missing libraries `libsvml` (Short vector math library) and `libimf` (Intel Math Library)

**Review by marcfehling — APPROVED** [review]

**Comment by marcfehling** [comment]
Didn't do the trick :(
Still:
```
1/7 Test #2: affinity.release .................***Failed    6.86 sec
/home/runner/work/dealii/dealii/tests/quick_tests/affinity.cc: In function ‘bool getaffinity(unsigned int&, unsigned int&)’:
/home/runner/work/dealii/dealii/tests/quick_tests/affinity.cc:53:51: error: dereferencing type-punned pointer will break strict-aliasing rules [-Werror=strict-aliasing]
   mask = *reinterpret_cast<unsigned int *>(&my_set);
                                                   ^
cc1plus: all warnings being treated as errors
```

**Comment by drwells** [comment]
`cpu_set_t` is a `struct`:

```c
typedef struct
{
  __cpu_mask __bits[__CPU_SETSIZE / __NCPUBITS];
} cpu_set_t;
```
so I would not expect casting this to an integer to make sense. We should rewrite this test to use the provided manipulation functions `CPU_COUNT_S` etc. instead.

**Comment by marcfehling** [comment]
Looks like there is something wrong with the order of linking with intel oneapi:
```
/home/runner/work/dealii/dealii/tests/quick_tests/affinity.debug: symbol lookup error: /opt/intel/oneapi/mkl/2022.0.2/lib/intel64/libmkl_gf_lp64.so.2: undefined symbol: mkl_blas_dgemm
```

Found this issue which might be related https://github.com/spack/spack/issues/7279.

**Review by marcfehling — APPROVED** [review]

**Comment by marcfehling** [comment]
I've opened a separate issue for the problems with the Intel oneAPI suite. Let's merge this!

**Comment by masterleinad** [comment]
I reverted the changes to the oneAPI build.

**Review by tamiko — APPROVED** [review]

**Pull request merged**